javascript/ql/src/semmle/javascript/ApiGraphs.qll

```diff
@@ -607,6 +607,16 @@ module API {
       rhs(_, nd) and
       result = nd.getALocalSource()
       or
+      // additional backwards step from `require('m')` to `exports` or `module.exports` in m
+      exists(Import imp | imp.getImportedModuleNode() = trackDefNode(nd, t.continue()) |
+        result.(ExportsAsSourceNode).getModule() = imp.getImportedModule()
+        or
+        exists(ModuleAsSourceNode mod |
+          mod.getModule() = imp.getImportedModule() and
+          result = mod.(DataFlow::SourceNode).getAPropertyRead("exports")
```

Comment:
Should there also be a case that steps to `getAPropertySource("exports")` to handle assignments to `module.exports`?

From the PR description it sounds like that case was already working, but just thought I'd check.

Reply:
I think that already works, but let me add a test for it.

Reply:
I have added a test which, to my not inconsiderable surprise, passed.